src/main/java/org/gridsuite/loadflow/server/computation/service/AbstractWorkerService.java

```diff
@@ -169,27 +171,46 @@ public Consumer<Message<String>> consumeCancel() {
+    protected void logOnRun(R runContext) {
         LOGGER.info("Run {} computation ...", getComputationType());
+    }
+
+    /**
+     * if this computation needs extra run data, add them to the run context here
+     */
+    protected void enrichRunContext(R runContext) { }
 
-        R context = resultContext.runContext;
-        String provider = context.getProvider();
+    /**
+     * if this computation needs extra report operations, override this function and do them here
+     */
+    protected void reportSpecificOperations(R runContext, Reporter reporter) { }
+
+    protected S run(R runContext, UUID resultUuid) throws Exception {
+        logOnRun(runContext);
+
+        enrichRunContext(runContext);
+
+        String provider = runContext.getProvider();
         AtomicReference<Reporter> rootReporter = new AtomicReference<>(Reporter.NO_OP);
         Reporter reporter = Reporter.NO_OP;
-        if (context.getReportContext() != null) {
-            final String reportType = context.getReportContext().getReportType();
-            String rootReporterId = context.getReportContext().getReportName() == null ? reportType : context.getReportContext().getReportName() + "@" + reportType;
+
+        if (runContext.getReportContext().getReportId() != null) {
+            final String reportType = runContext.getReportContext().getReportType();
+            String rootReporterId = runContext.getReportContext().getReportName() == null ? reportType : runContext.getReportContext().getReportName() + "@" + reportType;
             rootReporter.set(new ReporterModel(rootReporterId, rootReporterId));
             reporter = rootReporter.get().createSubReporter(reportType, String.format("%s (%s)", reportType, provider), "providerToUse", provider);
             // Delete any previous computation logs
-            observer.observe("report.delete", context, () -> reportService.deleteReport(context.getReportContext().getReportId(), reportType));
+            observer.observe("report.delete",
+                    runContext, () -> reportService.deleteReport(runContext.getReportContext().getReportId(), reportType));
         }
 
-        CompletableFuture<S> future = runAsync(network, context, provider, reporter, resultContext.getResultUuid());
+        Network network = getNetwork(runContext.getNetworkUuid(), runContext.getVariantId());
+        CompletableFuture<S> future = runAsync(network, runContext, provider, reporter, resultUuid);
 
-        S result = future == null ? null : observer.observeRun("run", context, future::get);
-        if (context.getReportContext().getReportId() != null) {
-            observer.observe("report.send", context, () -> reportService.sendReport(context.getReportContext().getReportId(), rootReporter.get()));
+        S result = future == null ? null : observer.observeRun("run", runContext, future::get);
+        if (runContext.getReportContext().getReportId() != null) {
```

Comment:
Add a `postRun` method and remove reportSpecificOperations

Reply:
Done.